runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -70,17 +76,43 @@
 	@Inject
 	private SkillIconManager skillIconManager;
 
-	private NavigationButton navButton;
-	private XpPanel xpPanel;
+	@Inject
+	private XpTrackerConfig xpTrackerConfig;
 
-	private final XpState xpState = new XpState();
+	@Inject
+	private XpState xpState;
 
+	@Inject
+	private XpPauseState xpPauseState;
+
+	private NavigationButton navButton;
+	private XpPanel xpPanel;
 	private WorldResult worlds;
 	private XpWorldType lastWorldType;
 	private String lastUsername;
+	private Instant lastTick;
 
 	private final XpClient xpClient = new XpClient();
 
+
+	@Provides
+	XpTrackerConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(XpTrackerConfig.class);
+	}
+
+	@Provides
+	XpState provideInternalState()
+	{
+		return new XpState();
+	}
+
+	@Provides
+	XpPauseState provideXpPauseState(XpTrackerConfig xpTrackerConfig, Client client)
```

Comment:
I dont think this is correct. The XpPauseState looks like it should not need to have access to entire client and entire config. I would rather just propagate to it what it really needs and not use guice for creating it at all

Reply:
I need:
1. Client game state
2. Access to all current skills XP values
3. Configuration settings on pausing (That's currently the only thing in the configuration)

Reply:
An alternate route is to include these on the tick function, rather than at construction.